core/services/relay/evm/write_target.go

```diff
@@ -69,7 +81,192 @@ func NewWriteTarget(ctx context.Context, relayer *Relayer, chain legacyevm.Chain
 		return nil, err
 	}
 
-	return targets.NewWriteTarget(logger.Named(lggr, "WriteTarget"), id, cr, cw, config.ForwarderAddress().String(), gasLimitDefault), nil
+	pollPeriod, err := commonconfig.NewDuration(2 * time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create poll period: %w", err)
+	}
+
+	// TODO: Unsure what the timeout should be, I don't see an EVM config that corresponds to this.
+	timeout, err := commonconfig.NewDuration(30 * time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create timeout: %w", err)
+	}
+
+	chainInfo, err := getChainInfo(chain.ID().Uint64())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chain info: %w", err)
+	}
+
+	beholder, err := writetarget.NewMonitor(ctx, lggr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Aptos WT monitor client: %+w", err)
+	}
+
+	opts := writetarget.WriteTargetOpts{
+		ID:     id,
+		Logger: lggr,
+		Config: writetarget.Config{
+			ConfirmerPollPeriod: pollPeriod,
+			ConfirmerTimeout:    timeout,
+		},
+		ChainInfo:        chainInfo,
+		Beholder:         beholder,
+		ChainService:     chain,
+		ContractReader:   cr,
+		ChainWriter:      cw,
+		ConfigValidateFn: evaluate,
+		NodeAddress:      "nil",
+		ForwarderAddress: config.ForwarderAddress().String(),
+		TargetStrategy:   NewEVMTargetStrategy(cr, cw, config.ForwarderAddress().String(), gasLimitDefault, lggr),
+	}
+
+	return writetarget.NewWriteTarget(opts), nil
+}
+
+type Inputs struct {
+	SignedReport types.SignedReport
+}
+
+type Request struct {
+	Metadata capabilities.RequestMetadata
+	Config   Config
+	Inputs   Inputs
+}
+
+func getEVMRequest(rawRequest capabilities.CapabilityRequest) (
+	Request, error) {
+	var r Request
+	r.Metadata = rawRequest.Metadata
+
+	if rawRequest.Config == nil {
+		return Request{}, errors.New("missing config field")
+	}
+
+	if err := rawRequest.Config.UnwrapTo(&r.Config); err != nil {
+		return Request{}, err
+	}
+
+	if !common.IsHexAddress(r.Config.Address) {
+		return Request{}, fmt.Errorf("'%v' is not a valid address", r.Config.Address)
+	}
+
+	if rawRequest.Inputs == nil {
+		return Request{}, errors.New("missing inputs field")
+	}
+
+	// required field of target's config in the workflow spec
+	signedReport, ok := rawRequest.Inputs.Underlying[writetarget.KeySignedReport]
+	if !ok {
+		return Request{}, fmt.Errorf("missing required field %s", writetarget.KeySignedReport)
+	}
+
+	if err := signedReport.UnwrapTo(&r.Inputs.SignedReport); err != nil {
+		return Request{}, err
+	}
+	return r, nil
+}
+
+func evaluate(rawRequest capabilities.CapabilityRequest) (receiver string, err error) {
```

Comment:
These two functions don't make much sense.

- `WriteTargetOps.ConfigValidateFn` should just validate the config, which here is the Ethereum address and gasLimit iirc
- The rest of the code is general and can be moved as part of the framework, is there anything EVM specific there? Some of these validations probably already happen as part of the WT framework processing, right?

Reply:
A lot of this validation is actually EVM specific, because A) it uses the `ReportV1Metadata` field which is not included in the aptos request as far as I can tell and B) it validates the EVM specific Request type. I don't see an issue leaving it as is for now and revisiting later as part of the greater generalization effort.